src/components/PostDetails.tsx

```diff
@@ -1,117 +1,147 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const { id, title, body } = post;
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [commentsLoadError, setCommentsLoadError] = useState('');
+  const [commentsDelError, setCommentsDelError] = useState<number[]>([]);
+  const [addNewComment, setAddNewComment] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    if (commentsLoadError) {
+      setCommentsLoadError('');
+    }
+
+    if (comments) {
+      setComments(null);
+    }
+
+    if (addNewComment) {
+      setAddNewComment(false);
+    }
+
+    client.get<Comment[]>(`/comments?postId=${id}`)
+      .then(setComments)
+      .catch(() => setCommentsLoadError("Can't load comments for current post"))
+      .finally(() => setLoading(false));
+  }, [post]);
+
+  function handleDeleteComment(commentId: number) {
+    client.delete(`/comments/${commentId}`)
+      .then(() => {
+        setComments(
+          current => current?.filter(item => item.id !== commentId) || null,
+        );
+        setCommentsDelError([]);
+      })
+      .catch(() => setCommentsDelError(current => [...current, commentId]));
+  }
+
+  const handleAddNewComment = (newComment: Comment) => {
+    setComments(current => (current ? [...current, newComment] : [newComment]));
+  };
```

Comment:
Use a consistent approach.

Or arrow functions, or function declaration

Reply:
https://github.com/mate-academy/react_dynamic-list-of-posts/pull/901#discussion_r1300277345

          <select>             
            <option value="0" disabled>Choose a user...</option>
            <option value="1">Option1</option>
            <option value="2" >Option2</option>
          </select>